core/frontend/src/Viewport.ts

```diff
@@ -1588,30 +1599,35 @@ export abstract class Viewport implements IDisposable {
 
     Camera.validateLensAngle(lensAngle);
 
-    if (view.isCameraOn)
-      return view.lookAtUsingLensAngle(view.getEyePoint(), view.getTargetPoint(), view.getYVector(), lensAngle);
-
-    // We need to figure out a new camera target. To do that, we need to know where the geometry is in the view.
-    // We use the depth of the center of the view for that.
-    let depthRange = this.determineVisibleDepthRange();
-    if (undefined === depthRange || Geometry.isAlmostEqualNumber(depthRange.minimum, depthRange.maximum))
-      depthRange = { minimum: 0, maximum: 1 };
-
-    const middle = depthRange.minimum + ((depthRange.maximum - depthRange.minimum) / 2.0);
-    const corners = [
-      new Point3d(0.0, 0.0, middle), // lower left, at target depth
-      new Point3d(1.0, 1.0, middle), // upper right at target depth
-      new Point3d(0.0, 0.0, depthRange.maximum), // lower left, at closest npc
-      new Point3d(1.0, 1.0, depthRange.maximum), // upper right at closest
-    ];
-
-    this.npcToWorldArray(corners);
-
-    const eye = corners[2].interpolate(0.5, corners[3]); // middle of closest plane
-    const target = corners[0].interpolate(0.5, corners[1]); // middle of halfway plane
-    const backDist = eye.distance(target) * 2.0;
-    const frontDist = view.minimumFrontDistance();
-    return view.lookAtUsingLensAngle(eye, target, view.getYVector(), lensAngle, frontDist, backDist);
+    let status;
+    if (view.isCameraOn) {
+      status = view.lookAtUsingLensAngle(view.getEyePoint(), view.getTargetPoint(), view.getYVector(), lensAngle);
+    } else {
+      // We need to figure out a new camera target. To do that, we need to know where the geometry is in the view.
+      // We use the depth of the center of the view for that.
+      let depthRange = this.determineVisibleDepthRange();
+      if (undefined === depthRange || Geometry.isAlmostEqualNumber(depthRange.minimum, depthRange.maximum))
+        depthRange = { minimum: 0, maximum: 1 };
+
+      const middle = depthRange.minimum + ((depthRange.maximum - depthRange.minimum) / 2.0);
+      const corners = [
+        new Point3d(0.0, 0.0, middle), // lower left, at target depth
+        new Point3d(1.0, 1.0, middle), // upper right at target depth
+        new Point3d(0.0, 0.0, depthRange.maximum), // lower left, at closest npc
+        new Point3d(1.0, 1.0, depthRange.maximum), // upper right at closest
+      ];
+
+      this.npcToWorldArray(corners);
+
+      const eye = corners[2].interpolate(0.5, corners[3]); // middle of closest plane
+      const target = corners[0].interpolate(0.5, corners[1]); // middle of halfway plane
+      const backDist = eye.distance(target) * 2.0;
+      const frontDist = view.minimumFrontDistance();
+      status = view.lookAtUsingLensAngle(eye, target, view.getYVector(), lensAngle, frontDist, backDist);
+    }
+
+    this.setupFromView();

```

Comment:
Should setupFromView only be called if lookAt returned ViewStatus.Success?

Reply:
I had thought lookAt might make some intermediate changes before returning an error but I now looked through  it and confirmed it will not. I'll change it.

Reply:
Changed.